packages/bloc/lib/src/bloc.dart

```diff
@@ -157,15 +165,15 @@ abstract class Bloc<Event, State> extends Stream<State> implements Sink<Event> {
   /// ```
   Stream<State> transformStates(Stream<State> states) => states;
 
-  void _bindStateSubject() {
+  void _bindEventsToStates() {
     Event currentEvent;
```

Comment:
General comment: It feels like this could lead to bugs where `asyncExpand` is not used in `transformEvents`, and therefore events will be delivered to the `forEach` that might respond to different events.

Reply:
Hmm yeah good point. Haven't run into that case so far but I see what you're saying.

Reply:
thoughts on https://github.com/felangel/bloc/pull/838?